### PR TITLE
Vt wallet connect 131109

### PR DESCRIPTION
**Pull request by tien7668**

### CODE REVIEW

File: src/js/containers/ImportAccount/ImportByWalletConnect.js [hunk 1]
```diff
@@ -42,23 +43,52 @@ export default class ImportByWallletConnect extends React.Component {
     async connect(e) {
         var wallet = this.state.wallet
         try {
-            var address = await wallet.getAddress()
-            this.setState({                
+            var {address, chainId} = await wallet.getAddress()
```
Comment (andrewsource147) [inline src/js/containers/ImportAccount/ImportByWalletConnect.js]: Please dont add chainId in here, seperate 2 function: getAddress(), getId(). Check getId() first before proceed with address

File: src/js/containers/ImportAccount/ImportByWalletConnect.js [hunk 2]
```diff
@@ -42,23 +43,52 @@ export default class ImportByWallletConnect extends React.Component {
...
             this.props.closeParentModal();
             this.closeModal();
-            this.props.dispatch(importNewAccount(address.toLowerCase(),
-                WalletType,
-                null,
-                this.props.ethereum,
-                this.props.tokens, null, null, "Wallet Connect"))
+
+            const networkId = BLOCKCHAIN_INFO.networkId
+            this.errorHandling(chainId, networkId, () => {
```
Comment (andrewsource147) [inline src/js/containers/ImportAccount/ImportByWalletConnect.js]: Please avoid to use callback function

File: src/js/containers/ImportAccount/ImportByWalletLink.js [hunk 3]
```diff
@@ -1,13 +1,14 @@
...
 
-import { importNewAccount, throwError } from "../../actions/accountActions"
-
 import { getTranslate } from 'react-localize-redux'
 
 import {getWallet} from "../../services/keys"
 
-
+import BLOCKCHAIN_INFO from "../../../../env";
+import {call} from "redux-saga/effects";
```
Comment (andrewsource147) [inline src/js/containers/ImportAccount/ImportByWalletLink.js]: Why having this import?

File: src/js/containers/ImportAccount/ImportByWalletLink.js [hunk 4]
```diff
@@ -36,20 +37,48 @@ export default class ImportByWallletLink extends React.Component {
...
             this.props.closeParentModal();
-            this.props.dispatch(importNewAccount(address.toLowerCase(),
-                WalletType,
-                null,
-                this.props.ethereum,
-                this.props.tokens, null, null, "Wallet Link"))
-                
+            const chainId = await wallet.getChainId()
+            const networkId = BLOCKCHAIN_INFO.networkId
+            this.errorHandling(chainId, networkId, () => {
```
Comment (andrewsource147) [inline src/js/containers/ImportAccount/ImportByWalletLink.js]: Still callback here?

File: src/js/containers/ImportAccount/ImportByWalletLink.js [hunk 5]
```diff
@@ -36,20 +37,48 @@ export default class ImportByWallletLink extends React.Component {
         try {
             var address = await wallet.getAddress()
             this.props.closeParentModal();
-            this.props.dispatch(importNewAccount(address.toLowerCase(),
-                WalletType,
-                null,
-                this.props.ethereum,
-                this.props.tokens, null, null, "Wallet Link"))
-                
+            const chainId = await wallet.getChainId()
```
Comment (andrewsource147) [inline src/js/containers/ImportAccount/ImportByWalletLink.js]: For walletlink, dont need to check chainId

Comment (ensete) [reply src/js/containers/ImportAccount/ImportByWalletLink.js]: I removed the logic for checking chainId

File: src/js/containers/ImportAccount/ImportByWalletLink.js (hunk 4 above)
Comment (ensete) [reply src/js/containers/ImportAccount/ImportByWalletLink.js]: Sorry, my bad

File: src/js/containers/ImportAccount/ImportByWalletLink.js (hunk 3 above)
Comment (ensete) [reply src/js/containers/ImportAccount/ImportByWalletLink.js]: Redundant code

File: src/js/containers/ImportAccount/ImportByWalletConnect.js (hunk 2 above)
Comment (ensete) [reply src/js/containers/ImportAccount/ImportByWalletConnect.js]: Removed

File: src/js/containers/ImportAccount/ImportByWalletConnect.js (hunk 1 above)
Comment (ensete) [reply src/js/containers/ImportAccount/ImportByWalletConnect.js]: I already separated them into 2 functions

**Pull request merged**